### ISSUE TITLE
Multiple examples indicate "[DefaultInstall]-based INF cannot be processed as Primitive"

### ISSUE THREAD

**Issue by dovholuknf** [microsoft/Windows-driver-samples#366]
I am new to windows driver development and I have been trying to learn my way using the samples in the network folder.  Yesterday I made the "mistake" of upgrading visual studio and now the drivers I am using are reporting 

`C:\git\external\windows-driver-samples\network\trans\inspect\sys\inspect.inf : error 1420: [DefaultInstall]-based INF cannot be processed as Primitive.`
and 
`
1>C:\git\external\Windows-driver-samples\network\trans\ddproxy\sys\ddproxy.inf : error 1420: [DefaultInstall]-based INF cannot be processed as Primitive.`

This page discusses the issue: https://docs.microsoft.com/en-us/windows-hardware/drivers/develop/creating-a-primitive-driver however the samples are not updated at this time.

**Comment by mutterbucket** [comment]
A solution that worked for me is to right click the driver project, and in the Driver Settings section change the Target Platform from Universal to Desktop.

**Comment by dovholuk** [comment]
> A solution that worked for me is to right click the driver project, and in the Driver Settings section change the Target Platform from Universal to Desktop.

This allows the project to build for me - but it won't actually deploy to the target machine. For me - the errors are just turned into warnings.

After you made that change - are you able to deploy the app? I get:
`Extracting service name from inf "C:\git\external\windows-driver-samples\network\trans\inspect\sys\x64\Debug\inspect\inspect.inf" Failed:`

**Comment by dovholuknf** [comment]
A solution that worked for me is to right click the driver project, and in the Driver Settings section change the Target Platform from Universal to Desktop.

This allows the project to build for me - but it won't actually deploy to the target machine. For me - the errors are just turned into warnings.

After you made that change - are you able to deploy the app? I get:
Extracting service name from inf "C:\git\external\windows-driver-samples\network\trans\inspect\sys\x64\Debug\inspect\inspect.inf" Failed:

**Comment by mutterbucket** [comment]
I can run them on a VMWare test machine with a serial connection to WinDbg and driver signature enforcement disabled, that has been the most reliable setup I can establish right now.

**Comment by estrandb** [comment]
https://docs.microsoft.com/en-us/windows-hardware/drivers/develop/creating-a-primitive-driver#legacy-compatibility

I was getting this error as well. As described in the above documentation, I decorated the DefaultInstall and DefaultUninstall attributes with .NTx86 and added LegacyUninstall = 1 to the DefaultUninstall.NTx86 section. It builds now. I don't know if it works yet though ;)

**Comment by rakendrathapa** [comment]
This behavior introduced with Windows 10 version 1903 and later:
Look here: https://docs.microsoft.com/en-us/windows-hardware/drivers/develop/creating-a-primitive-driver

**Comment by msojocs** [comment]
```

;;
;; Default install sections
;;

[DefaultInstall.NTamd64]
OptionDesc          = %SimRepServiceDesc%
CopyFiles           = SimRep.DriverFiles

[DefaultInstall.NTamd64.Services]
AddService          = %SimRepServiceName%,,SimRep.Service

;;
;; Default uninstall sections
;;

[DefaultUninstall.NTamd64]
LegacyUninstall=1
DelFiles   = SimRep.DriverFiles
DelReg     = SimRep.DelRegistry


[DefaultUninstall.NTamd64.Services]
DelService = SimRep,0x204

```
![image](https://github.com/microsoft/Windows-driver-samples/assets/20937135/c7632cb0-e21e-4d95-8173-9146f77e5f43)

**Comment by MrRhuezzler** [comment]
> A solution that worked for me is to right click the driver project, and in the Driver Settings section change the Target Platform from Universal to Desktop.
> 
> This allows the project to build for me - but it won't actually deploy to the target machine. For me - the errors are just turned into warnings.
> 
> After you made that change - are you able to deploy the app? I get: Extracting service name from inf "C:\git\external\windows-driver-samples\network\trans\inspect\sys\x64\Debug\inspect\inspect.inf" Failed:

I am trying to kernel debug a driver, I am running into the same issue. Although I can ignore the errors and I am able to kernel debug but I need to install the driver manually in the test system after attaching the kernel debugger from visual studio.

It would be nice if someone has solved this.

**Comment by v-junyli** [comment]
We’re currently improving our issue management process to keep this repository healthy and easier to maintain. This issue is being closed because there hasn’t been any activity in the past 6 months.

If this is still a problem for you, please feel free to open a new issue with the latest details — we’ll be happy to take another look.

Thanks for your understanding and for helping us keep the repo in good shape!

**Comment by dovholuknf** [comment]
yes, "just closing issues" is always a major improvement. i love it when repos do that. "super"
